plugins/modules/k8s_taint.py

```diff
@@ -0,0 +1,279 @@
+                type: str
+            effect:
+                description:
+                    - The effect of the taint on Pods that do not tolerate the taint.
+                    - Required when I(state=present).
+                type: str
+                choices: [ NoSchedule, NoExecute, PreferNoSchedule ]
+    overwrite:
+        description:
+            - If true, allow taints to be overwritten, otherwise reject taint updates that overwrite existing taints.
+        required: false
+        default: false
+        type: bool
+requirements:
+  - python >= 3.6
+  - kubernetes >= 12.0.0
+"""
+
+EXAMPLES = r"""
+- name: Taint node "foo"
+  kubernetes.core.k8s_taint:
+    state: present
+    name: foo
+    taints:
+        - effect: NoExecute
+          key: "key1"
+
+- name: Taint node "foo"
+  kubernetes.core.k8s_taint:
+    state: present
+    name: foo
+    taints:
+        - effect: NoExecute
+          key: "key1"
+          value: "value1"
+        - effect: NoSchedule
+          key: "key1"
+          value: "value1"
+
+- name: Remove taint from "foo".
+  kubernetes.core.k8s_taint:
+    state: absent
+    name: foo
+    taints:
+        - effect: NoExecute
+          key: "key1"
+          value: "value1"
+"""
+
+RETURN = r"""
+result:
+    description:
+        - The tainted Node object.
+    returned: success
+    type: complex
+    contains:
+        api_version:
+            description: The versioned schema of this representation of an object.
+            returned: success
+            type: str
+        kind:
+            description: Represents the REST resource this object represents.
+            returned: success
+            type: str
+        metadata:
+            description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+            returned: success
+            type: complex
+        spec:
+            description: Specific attributes of the object. Will vary based on the I(api_version) and I(kind).
+            returned: success
+            type: complex
+        status:
+            description: Current status details for the object.
+            returned: success
+            type: complex
+"""
+
+import copy
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+from ansible_collections.kubernetes.core.plugins.module_utils.common import (
+    K8sAnsibleMixin,
+    get_api_client,
+)
+from ansible_collections.kubernetes.core.plugins.module_utils.args_common import (
+    AUTH_ARG_SPEC,
+)
+
+try:
+    from kubernetes.client.api import core_v1_api
+    from kubernetes.dynamic.exceptions import ApiException
+except ImportError:
+    # ImportError are managed by the common module already.
+    pass
+
+
+def _equal_dicts(a, b):
+    ignore_keys = ("time_added", "value")
+    ka = set(a).difference(ignore_keys)
+    kb = set(b).difference(ignore_keys)
+
+    return all(a[k] == b[k] for k in ka if k in kb)
+
```

Comment:
```suggestion
def _equal_dicts(a, b):
    ignore_keys = ("time_added", "value")
    ka = set(a).difference(ignore_keys)
    kb = set(b).difference(ignore_keys)

    return ka == kb and all(a[k] == b[k] for k in ka if k in kb)

```

seems that this method is returning false positive for 
```
a = {'a': 1, 'b': 2}
b = {'a': 1}
```

Reply:
I think the function name is probably just a little misleading, but I can't think of a better one. A taint of `{"key": "foo"}` should remove any existing taints with `key=foo` regardless what their value/effect are, so in this case they should be considered "equal enough". I believe this is working correctly, though it would be a good idea to add a few more tests for these kinds of cases.

Reply:
The input won't be in this way.
```
a = {'a': 1, 'b': 2}
b = {'a': 1}
```
The input would always has the same size (for this reason I remove some fields). a and b would always be a combination of {"effect": "NoExecute", "key": "key1"}, so the function works as expected.

Reply:
Yeah, like this compact way. Thanks.